### ISSUE TITLE
Assigning Thread to static property of the class during constructor causes segfault on thread start

### ISSUE THREAD

**Issue by dktapps** [krakjoe/pthreads#704]
Environment
==========

  * PHP: 7.1.5
  * pthreads: 3.1.7dev (7004081)
  * OS: Ubuntu 16.04

Summary
======
The below code sample worked fine with php 7.0.19 and pthreads 3.1.6, segfaults on 7.1.5 and pthreads 3.1.7dev.

Reproducing Code
==============
Broken:
```
<?php
class test extends Thread{
	public static $me = null;

	public function __construct(){
		static::$me = $this;
	}

	public function run(){
		sleep(5);
	}
}

$t = new test();
$t->start();
$t->join();
```

Working:
```
<?php
class test extends Thread{
	public static $me = null;

	public function run(){
		sleep(5);
	}
}

$t = new test();
test::$me = $t;
$t->start();
$t->join();
```

Expected output
=============
Nothing

Actual Output
============
`Segmentation fault (core dumped)`

**Comment by SirSnyder** [comment]
This got fixed with https://github.com/krakjoe/pthreads/pull/683. Check out my fork https://github.com/SirSnyder/pthreads/tree/php/71

**Comment by dktapps** [comment]
@SirSnyder I tested your fork and it does indeed resolve the issue. Thanks very much.

**Comment by tpunt** [comment]
Fixed in 96859b1fe82f1a7fa804e516ae236c128a360437
